### PR TITLE
Add Github presubmit check for minimum_supported_agent_version author

### PR DESCRIPTION
**Pull request by avilevy18**
Add GitHub presubmit tests to check whether minimum_supported_agent_version is specified. Fail the tests if it is specified and the author of the PR is not our bot stackdriver-instrumentation-release.

http://b/238471194

### CODE REVIEW

File: .github/workflows/pr_presubmit.yaml
Comment (sophieyfang) [inline .github/workflows/pr_presubmit.yaml:21]: What does `github.event.pull_request.labels.*.name` usually contain? Is it the pull request author name? 

Could we add a comment to explain what it refers to?

File: ci/minimum_supported_agent_version_test.go
Comment (ridwanmsharif) [inline ci/minimum_supported_agent_version_test.go:30]: Is this the check for whether the stackdriver-instrumentation-release bot is making the change? I'm not familiar with what the labels are populated with but can we make this check more strict? Checking for the bot's github username or failing that, adding some comments here about what the labels are expected to have.

**Review by ridwanmsharif — COMMENTED** [review]
Thanks for working on this, it will help so much! I left a few comments, mostly minor nits. I did have a question though - how are we testing this change before merging?

File: ci/minimum_supported_agent_version_test.go
Comment (sophieyfang) [inline ci/minimum_supported_agent_version_test.go:20]: Hmmm. are we using the labels to tell who the author is? The pull request trigger by post release script doesn't attach any labels at this current moment

File: ci/minimum_supported_agent_version_test.go
Comment (sophieyfang) [inline ci/minimum_supported_agent_version_test.go:69]: Trying to understand those lines. It could be my problem..

So `var emptyMetadata metadata.MinimumSupportedAgentVersion` sets a by default null or empty `MinimumSupportedAgentVersion`
Then the next if statement compares if the parsed added file has the same `MinimumSupportedAgentVersion` with the empty one?  I guess you were trying to check if the parsed one is not empty then good. If it's empty and meaningless, it shall fail the test?

File: ci/minimum_supported_agent_version_test.go
Comment (avilevy18) [reply ci/minimum_supported_agent_version_test.go:30]: A "release" label will be added to the PR, the flag `labels` take in all the labels for the PR delimited by spaces.The stackdriver-instrumentation-release bot has been configured to automatically add the label. That being said any user who needs to edit the `minimum_supported_agent_version` would need to add the label manually to be able to do any required fixes.

File: ci/minimum_supported_agent_version_test.go
Comment (sophieyfang) [reply ci/minimum_supported_agent_version_test.go:69]: I think i get it now. If the PR label contains release, then always return early and true. If not, and it contains changes of `MinimumSupportedAgentVersion` in either added or modified files, fail it. Though i am still not sure if a developer put below in the metadata.yaml file, does that count as emptyMetadata file and the PR will pass the test which it should not
```
...
minimum_supported_agent_version:
....
```

File: ci/minimum_supported_agent_version_test.go
Comment (igorpeshansky) [reply ci/minimum_supported_agent_version_test.go:20]: We will change the release script to add the appropriate labels. This approach allows a maintainer to fix up `minimum_supported_agent_version` in case of an error, which would be very cumbersome if we checked the author instead.

File: ci/minimum_supported_agent_version_test.go
Comment (ridwanmsharif) [reply ci/minimum_supported_agent_version_test.go:30]: I see that makes sense to me now. Can we add this as a comment as well, mentioning that the bot will configure this label and maintainers can use the label if they want to fix up `minimum_supported_agent_version` in case of an error

Comment (avilevy18) [reply ci/minimum_supported_agent_version_test.go:34]: Changed in: [63bfb6b](https://github.com/GoogleCloudPlatform/ops-agent/pull/775/commits/63bfb6bfccdaef27380cd30c086a1590ea925d47)

File: ci/minimum_supported_agent_version_test.go
Comment (avilevy18) [reply ci/minimum_supported_agent_version_test.go:112]: Changed in: [63bfb6b](https://github.com/GoogleCloudPlatform/ops-agent/pull/775/commits/63bfb6bfccdaef27380cd30c086a1590ea925d47)

File: .github/workflows/pr_presubmit.yaml
Comment (avilevy18) [reply .github/workflows/pr_presubmit.yaml:21]: For the comment would a link to the github documentation suffice?
https://docs.github.com/en/rest/issues/labels

File: ci/minimum_supported_agent_version_test.go
Comment (avilevy18) [reply ci/minimum_supported_agent_version_test.go]: Changed in: [63bfb6b](https://github.com/GoogleCloudPlatform/ops-agent/pull/775/commits/63bfb6bfccdaef27380cd30c086a1590ea925d47)

File: ci/minimum_supported_agent_version_test.go
Comment (avilevy18) [reply ci/minimum_supported_agent_version_test.go]: Changed in: [63bfb6b](https://github.com/GoogleCloudPlatform/ops-agent/pull/775/commits/63bfb6bfccdaef27380cd30c086a1590ea925d47)

File: ci/minimum_supported_agent_version_test.go
Comment (sophieyfang) [reply ci/minimum_supported_agent_version_test.go:21]: Right, good thought!

File: ci/minimum_supported_agent_version_test.go
Comment (ridwanmsharif) [inline ci/minimum_supported_agent_version_test.go:112]: Do we need this method anymore? We could just use the new `UnmarshalAndValidate` method in here too no?

Comment (avilevy18) [reply ci/minimum_supported_agent_version_test.go:111]: Changed in: [8f59aea](https://github.com/GoogleCloudPlatform/ops-agent/pull/775/commits/8f59aea4d1f85a412940447e975a2bb767cf36f1)

File: ci/minimum_supported_agent_version_test.go
Comment (avilevy18) [reply ci/minimum_supported_agent_version_test.go]: Changed in: [95aa45e](https://github.com/GoogleCloudPlatform/ops-agent/pull/775/commits/95aa45e4655aa3a43fc8943102afaf47950eb7c7)

**Review by ridwanmsharif — APPROVED** [review]

**Pull request merged**
